### PR TITLE
Enabled Plugins Report

### PR DESCRIPTION
**Pull request by madebyrogal**
## Description

Changes proposed in this pull request:

- Report to Segment plugins RBAC configuration

## Testing

- In `internal/analytics/segment_reporter.go:36` set APIKey (you can find it here: https://app.segment.com/kubeshop-dev-botkube/sources/botkube_agent/settings/keys)
- Run botkube agent
- Open Segment Debuger https://app.segment.com/kubeshop-dev-botkube/sources/botkube_agent/debugger
- Check if you can see a new event `PluginConfig`

## Related issue(s)

- https://github.com/kubeshop/botkube-cloud/issues/836
- PR on botkube-docs (update privacy policy) https://github.com/kubeshop/botkube-docs/pull/308

### CODE REVIEW

File: cmd/botkube-agent/main.go [hunk 1]
```diff
@@ -152,6 +152,10 @@ func run(ctx context.Context) (err error) {
 	if err != nil {
 		return reportFatalError("while registering current identity", err)
 	}
+	err = analyticsReporter.ReportPluginsConfig(conf.Executors, conf.Sources)
+	if err != nil {
+		return reportFatalError("while reporting plugins configuration", err)
```
Comment (pkosiec) [inline cmd/botkube-agent/main.go:157]: This shouldn't be a fatal error - let's log it and continue.

File: internal/analytics/segment_reporter.go
Comment (pkosiec) [inline internal/analytics/segment_reporter.go:1]: Please update the tests in the `segment_reporter_test.go` file. They are golden-file based. Thanks!

File: internal/analytics/segment_reporter.go [hunk 2]
```diff
@@ -106,6 +113,25 @@ func (r *SegmentReporter) ReportBotEnabled(platform config.CommPlatformIntegrati
 	})
 }
 
+func (r *SegmentReporter) ReportPluginsConfig(executors map[string]config.Executors, sources map[string]config.Sources) error {
+	pluginsConfig := make(map[string]interface{})
+	for _, values := range executors {
+		for name, pluginValue := range values.Plugins {
+			if pluginValue.Enabled {
```
Comment (pkosiec) [inline internal/analytics/segment_reporter.go:120]: You can invert the condition and in that way make the code simpler (with less indentation).

```go
if pluginValue.Disabled {
    continue
}
```

Change it in both places.

File: internal/analytics/segment_reporter.go [hunk 3]
```diff
@@ -333,3 +359,11 @@ func (r *SegmentReporter) getNodeCount(ctx context.Context, k8sCli kubernetes.In
 
 	return workerNodesCount, controlPlaneNodesCount, nil
 }
+
+func generatePluginReport(plugin config.Plugin, pluginName string, pluginType plugin.Type) pluginReport {
+	return pluginReport{
+		Name: pluginName,
+		Type: pluginType,
+		RBAC: plugin.Context.RBAC,
```
Comment (pkosiec) [inline internal/analytics/segment_reporter.go:367]: As I mentioned earlier, we should anonymize the reporting. That means we should remove anything what could identify the user.
Like:
- rbac group name
- rbac user name

BTW I'd suggest to convert this to a method, to scope it to the Segment reporter.

File: cmd/botkube-agent/main.go
Comment (pkosiec) [inline cmd/botkube-agent/main.go:1]: Please also remember about a PR to docs to update the privacy policy as we'll collect enabled plugin names and anonymized RBAC configuration.

**Review by pkosiec — COMMENTED** [review]
Just a few comments after code review 🙂 Didn't test it yet.

File: cmd/botkube-agent/main.go
Comment (madebyrogal) [reply cmd/botkube-agent/main.go:1]: Here is PR https://github.com/kubeshop/botkube-docs/pull/308 (with fixed link to k8s event)

**Review by pkosiec — COMMENTED** [review]
Some minor comments, I'm about to test the changes in next few minutes and approve it 👍

File: internal/analytics/segment_reporter_test.go [hunk 4]
```diff
@@ -151,6 +151,105 @@ func TestSegmentReporter_ReportBotEnabled(t *testing.T) {
...
+				"botkube/kubernetes": {
+					Enabled: false,
+					Config:  "{}",
+					Context: config.PluginContext{
+						RBAC: &config.PolicyRule{
+							User: config.UserPolicySubject{
+								Type: config.EmptyPolicySubjectType,
+								Static: config.UserStaticSubject{
+									Value: "",
+								},
```
[the remark this answers is not in the document]
Comment (madebyrogal) [reply internal/analytics/segment_reporter_test.go:231]: This plugin is actually disabled but I know what you mean (changed in k9s rbac config ~#201)

**Review by pkosiec — APPROVED** [review]
🚀 Good job!

**Pull request merged**